sdks/java/core/src/main/java/org/apache/beam/sdk/metrics/DelegatingPerWorkerCounter.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.metrics;
+
+import java.io.Serializable;
+import org.apache.beam.sdk.annotations.Internal;
+
+/** Implementation of {@link Counter} that delegates to the instance for the current context. */
+@Internal
+public class DelegatingPerWorkerCounter implements Metric, Counter, Serializable {
+  private final MetricName name;
+  private final boolean processWideContainer;
+
+  public DelegatingPerWorkerCounter(MetricName name) {
+    this(name, false);
+  }
+
+  public DelegatingPerWorkerCounter(MetricName name, boolean processWideContainer) {
+    this.name = name;
+    this.processWideContainer = processWideContainer;
+  }
+
+  /** Increment the counter. */
+  @Override
+  public void inc() {
+    inc(1);
+  }
+
+  /** Increment the counter by the given amount. */
+  @Override
+  public void inc(long n) {
+    MetricsContainer container =
+        this.processWideContainer
+            ? MetricsEnvironment.getProcessWideContainer()
+            : MetricsEnvironment.getCurrentContainer();
+    if (container != null) {
+      container.getPerWorkerCounter(name).inc(n);
```

Comment:
Do we need a separate implementation of Delegating counter just for this line change?

Reply:
Extended DelegatingCounter to also handle perWorkerCounters.